main/src/ca/uwaterloo/flix/language/ast/ops/SimplifiedAstOps.scala

```diff
@@ -513,4 +515,291 @@ object SimplifiedAstOps {
     root
   }
 
+  def renameBoundVars(exp0: Expression, vars0: Map[Symbol.VarSym, Symbol.VarSym], labels0: Map[Symbol.LabelSym, Symbol.LabelSym])(implicit flix: Flix): Expression =
+    exp0 match {
+      case Expression.Unit => exp0
+      case Expression.True => exp0
+      case Expression.False => exp0
+      case Expression.Char(lit) => exp0
+      case Expression.Float32(lit) => exp0
+      case Expression.Float64(lit) => exp0
+      case Expression.Int8(lit) => exp0
+      case Expression.Int16(lit) => exp0
+      case Expression.Int32(lit) => exp0
+      case Expression.Int64(lit) => exp0
+      case Expression.BigInt(lit) => exp0
+      case Expression.Str(lit) => exp0
+
+      case Expression.Def(sym, tpe, loc) => Expression.Def(sym, tpe, loc)
+
+      case Expression.Var(sym, tpe, loc) =>
+        val s = vars0.getOrElse(sym, sym) // sym might not have been renamed e.g. if it's a free variable.
+        Expression.Var(s, tpe, loc)
+
+      case Expression.Lambda(args0, body, tpe, loc) =>
+        val args1 = args0.map(fparam => fparam.copy(sym = Symbol.freshVarSym(fparam.sym)))
+        val vars1 = args0.zip(args1).map {
+          case (oldParam, newParam) => oldParam.sym -> newParam.sym
+        }
+        val b = renameBoundVars(body, vars1.toMap ++ vars0, labels0)
+        Expression.Lambda(args1, b, tpe, loc)
+
+      case Expression.Apply(exp, args, tpe, loc) =>
+        val e = renameBoundVars(exp, vars0, labels0)
+        val as = args map (renameBoundVars(_, vars0, labels0))
```

Comment:
I believe Scala is going to deprecated `args map` with spacing, so we are slowly migrating to `args.map`.